src/nedap/utils/test/impl.cljc

```diff
@@ -64,22 +64,39 @@
      (defmethod cljs.test/report [::exit-code-reporter :end-run-tests] [summary]
        (set-exit-code-for! summary js/process))))
 
+(defmulti expect-matcher
+          "Given a symbol returns :assert-expr-sym, :pred-sym and :pred.
+
+           - `:assert-expr-sym` is used to construct the `is` form to preserve clojure.tes/assert-expr behaviour.
+           - `:pred` is used to assert validity on compile time.
+           - `:pred-sym` is used to assert validity on runtime.
+
+           These nuances exist for interoperability between clj and cljs runtime."
+          identity)
+
+(defmethod expect-matcher '= [_]
+  {:assert-expr-sym '=
+   :pred-sym `=
+   :pred =})
+
 (defn expect
-  [bodies {:keys [to-change from to] :as opts} clj?]
+  [bodies {:keys [to-change from to with] :as opts} clj?]
   {:pre [(spec/valid? boolean? clj?)]}
   (assert (seq bodies) "bodies can't be empty")
-  (assert (= #{:to-change :from :to} (set (keys opts))) (pr-str opts))
-  (assert (not (meta= [from to]))
-          (binding [*print-meta* true]
-            (str (pr-str from) " should be different from " (pr-str to))))
-  (assert (some? to-change) (pr-str to-change))
+  (assert (= #{:to-change :from :to} (set (keys (dissoc opts :with)))) (pr-str opts))
+  (let [{:keys [pred-sym pred assert-expr-sym]} (expect-matcher with)
+        is (if clj? 'clojure.test/is 'cljs.test/is)]
+    (assert (every? some? [pred-sym pred assert-expr-sym]))
```

Comment:
This doesn't report gracefully; better to place one `assert` call for each of `[pred-sym pred assert-expr-sym]`, maybe with an explanation string